sr_test_force_resolution/src/sr_test_force_resolution/sr_test_force_resolution.py

```diff
@@ -0,0 +1,451 @@
+            self.run()
+
+    def run(self):
+        CONST_EXIT_CHAR = 'Q'
+        CONST_POSITION_CHAR = 'P'
+        CONST_EFFORT_CHAR = 'E'
+        self.print_help()
+        in_string = raw_input("Please enter command:").upper()
+        if in_string == "":
+            return
+        input_commnad = in_string.split('_')
+        command_type = input_commnad[0].upper()
+        if CONST_EXIT_CHAR in command_type:
+            rospy.loginfo("Quitting...")
+            sys.exit(0)
+        joint = input_commnad[1].upper()
+        angle = input_commnad[2].lower()
+        if 'min' in angle:
+            angle = self._joint_ranges[joint.split('J')[0]][joint.split('J')[1]]['min']
+        elif 'max' in angle:
+            angle = self._joint_ranges[joint.split('J')[0]][joint.split('J')[1]]['max']
+        rospy.loginfo("type: %s joint: %s angle: %s", command_type, joint, angle)
+        if (command_type not in CONST_EXIT_CHAR and
+            command_type not in CONST_EFFORT_CHAR and
+                command_type not in CONST_POSITION_CHAR):
+            rospy.logerr("command type %s not recognised", command_type)
+            return
+        if (joint not in self.requested_joints and
+                joint not in 'ALL' and 'FF' in joint):
+            rospy.logerr("joint %s not recognised", joint)
+            return
+        if CONST_POSITION_CHAR in command_type:
+            if 'FF' in joint:
+                maximum = self._joint_ranges[joint.split('J')[0]][joint.split('J')[1]]['max']
+                minimun = self._joint_ranges[joint.split('J')[0]][joint.split('J')[1]]['min']
+                if float(angle) > maximum:
+                    rospy.logerr("angle outside range for %s: %s > %s", joint, angle, maximum)
+                    return
+                if float(angle) < minimun:
+                    rospy.logerr("angle outside range for %s: %s < %s", joint, angle, minimun)
+                    return
+        print "type: {type} joint: {joint} angle: {angle}".format(type=command_type, joint=joint, angle=angle)
+        rospy.loginfo("type: %s joint: %s angle: %s", command_type, joint, angle)
+        if [j for j in ['FF', 'RF', 'MF', 'LF', 'TH'] if j in joint]:
+            if command_type.upper() == CONST_POSITION_CHAR:
+                self.test_joint(joint, mode='position', value=angle)
+            elif command_type.upper() == CONST_EFFORT_CHAR:
+                file_prefix = raw_input("Name/describe the test (adds text to filename, optional)")
+                for i in range(0, 5):
+                    self.test_joint(joint, mode='PWM', value=str(float(angle)*1.0), prefix=file_prefix + '_plus')
+                    rospy.sleep(1)
+                    self.test_joint(joint, mode='PWM', value=str(float(angle)*-1.0), prefix=file_prefix + '_minus')
+                    rospy.sleep(1)
+        else:
+            lst = [j for j in self.requested_joints if str(joint[1]) in j and j[0]+j[1] not in self._fingers_with_j0]
```

Comment:
improve name, and also here would be good to wrap the long line in a function with clear name